specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -131,106 +131,124 @@ namespace AzureMessagingEventGrid {
+    @visibility("read")
+    eventSubscriptionName: string;
+  }
 
-  @doc("Publish Batch of Cloud Events to namespace topic.")
-  @route("/topics/{topicName}:publish", {shared: true})
-  @post op PublishBatchOfCloudEvents is Azure.Core.RpcOperation<{
-    @doc("content type")
-    @header("content-type")
-    contentType: "application/cloudevents-batch+json; charset=utf-8";
+  @doc("Publish operation Response")
+  model PublishResponse {}
 
-    @doc("Topic Name.")
-    @path
-    topicName: string;
 
-    @doc("Array of Cloud Events being published.")
-    @body
-    events: CloudEventEvent[];
-  }, OkResponse>;
-   
-  @doc("Receive Batch of Cloud Events from the Event Subscription.")
-  @route("/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive")
-  @post op ReceiveBatchOfCloudEvents is Azure.Core.RpcOperation<{
-    @doc("Topic Name.")
-    @path
-    topicName: string;
+  // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
 
-    @doc("Event Subscription Name.")
-    @path
-    eventSubscriptionName: string;
+  @doc("Publish Single Cloud Event to namespace topic.")
+  @route("/", {shared: true})
+  @action("publish")
+  @post op PublishCloudEvent is Azure.Core.ResourceAction<
+    TopicResource,
+    {
+      @doc("content type")
+      @header("content-type")
+      contentType: "application/cloudevents+json; charset=utf-8";
+
+      @doc("Single Cloud Event being published.")
+      @body 
+      event: CloudEventEvent;
+    },
+    PublishResponse
+    >;
 
-    @doc("Max Events count to be received.")
-    @query
-    maxEvents?: int32 = 1;
+  @doc("Publish Batch of Cloud Events to namespace topic.")
+  @route("/", {shared: true})
+  @action("publish")
+  @post op PublishBatchOfCloudEvents is Azure.Core.ResourceAction<
+    TopicResource,
+    {
+      @doc("content type")
+      @header("content-type")
+      contentType: "application/cloudevents-batch+json; charset=utf-8";
+
+      @doc("Array of Cloud Events being published.")
+      @body
+      events: CloudEventEvent[];
+    },
+    PublishResponse
+    >;
+
+  // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive?api-Version={apiVersion}&timeout=60&maxEvents={maxEvents}
 
-    @doc("Timeout value for receive operation in Seconds. Default is 60 seconds.")
-    @query
-    timeout?: int32 =60;
-  }, ReceiveResponse & CreatedResponse >;
+   
+  @doc("Receive Batch of Cloud Events from the Event Subscription.")
+  @action("receive")
+  @post op ReceiveBatchOfCloudEvents is Azure.Core.ResourceAction<
+    EventSubscription,
+    {
+      @doc("Max Events count to be received.")
+      @query
+      maxEvents?: int32 = 1;
+
+      @doc("Timeout value for receive operation in Seconds. Default is 60 seconds.")
+      @query
+      timeout?: int32 =60;
+    },
+    ReceiveResponse & CreatedResponse
```

Comment:
where CreatedResponse is defined? and why we needed both ReceivedResponse and CreatedResponse  here??

Reply:
CreatedResponse is a TypeSpec type that means "201". Once you service has been update per stewardship board guidelines to not return 201 anymore, this can go away.

Reply:
This is pre-service changes, currently this operation is returning a 201 along with ReceiveResponse, when the service fixes the return to be the 200 discussed in the APIReview meeting we can remove this. This is just to unblock the SDK team on generating/running code. 

CreatedResponse is Typespec.Http.CreatedResponse so it is within Typespec.http

Reply:
TypeSpec was not letting me to union 2 response types together since it duplicates statusCode, I can find a way around that if we want, but if we are going to change to 200 is it worthwhile to change it now as well?